utility/network.go

```diff
@@ -0,0 +1,118 @@
+package utility
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/retry"
+
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jwt"
+
+	"google.golang.org/api/androidpublisher/v3"
+)
+
+// CreateHTTPClient creates an HTTP client for the communication during the uploads.
+func CreateHTTPClient(jsonKeyPth string) (*http.Client, error) {
+	jsonKeyPth, isRemote, err := ParseURI(string(jsonKeyPth))
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare key path (%s), error: %s", jsonKeyPth, err)
+	}
+
+	if isRemote {
+		tmpDir, err := pathutil.NormalizedOSTempDirPath("__google-play-deploy__")
+		if err != nil {
+			return nil, fmt.Errorf("failed to create tmp dir, error: %s", err)
+		}
+
+		jsonKeySource := jsonKeyPth
+		jsonKeyPth = filepath.Join(tmpDir, "key.json")
+		if err := downloadFileWithRetry(jsonKeySource, jsonKeyPth, 3, 3); err != nil {
```

Comment:
NOTE: no need need to download to a file, byte[] would work just fine; find reusing jsonKeyPth a bit confusing. Realize this is not new code.

Reply:
Updated code.